docs/spec/concepts.rst

```diff
@@ -3,6 +3,336 @@
+if the set of values represented by ``A`` is a superset of the values
+represented by ``B``. The supertype relation is also transitive and reflexive.
+
+We also define an **equivalence** relation on fully static types: the types
+``A`` and ``B`` are equivalent (or "the same type") if and only if ``A`` is a
+subtype of ``B`` and ``B`` is a subtype of ``A``. This means that the set of
+values represented by ``A`` is both a superset and a subset of the values
+represented by ``B``, meaning ``A`` and ``B`` must represent the same set of
+values.
+
+We may describe a type ``B`` as "narrower" than a type ``A`` (or as a "proper
+subtype" of ``A``) if ``B`` is a subtype of ``A`` and ``B`` is not equivalent
+to ``A``. In the same scenario we can describe the type ``A`` as "wider" than
+``B``, or a "proper supertype" of ``B``.
+
+Nominal and structural types
+----------------------------
+
+For a type such as ``str`` (or any other class), which describes the set of
+values whose ``__class__`` is ``str`` or a direct or indirect subclass of it,
+subtyping corresponds directly to subclassing. A subclass ``MyStr`` of ``str``
+is a subtype of ``str``, because ``MyStr`` represents a subset of the values
+represented by ``str``. Such types can be called "nominal types" and this is
+"nominal subtyping."
+
+Other types (e.g. :ref:`Protocols` and :ref:`TypedDict`) instead describe a set
+of values by the types of their attributes and methods, or the types of their
+dictionary keys and values. These are called "structural types." A structural
+type may be a subtype of another type without any inheritance or subclassing
+relationship, simply because it shares all the attributes/methods (or
+keys/values) of the supertype, and adds additional requirements, thus
+representing a subset of the possible values. This is "structural subtyping."
+
+Although the means of specifying the set of values represented by the types
+differs, the fundamental concepts are the same for both nominal and structural
+types: the type represents a set of possible values and a subtype represents a
+subset of the values.
+
+Materialization
+---------------
+
+Since :ref:`Any` represents an unknown static type, it does not represent any
+known single set of values (it represents an unknown set of values). Thus it is
+not in the domain of the subtype, supertype, or equivalence relations on static
+types described above.
+
+To relate gradual types more generally, we define a **materialization**
+relation. Materialization transforms a "more dynamic" type to a "more static"
+type. Given a gradual type ``A``, if we replace zero or more occurrences of
+``Any`` in ``A`` with some gradual type (which can be different for each
+occurrence of ``Any``), the resulting gradual type ``B`` is a materialization
+of ``A``. (We can also materialize a :ref:`Callable` type by replacing ``...``
+with any type signature, and materialize ``tuple[Any, ...]`` by replacing it
+with a determinate-length tuple type.)
```

Comment:
Or by an indeterminate-length tuple type such as `tuple[int, ...]`, which is also a fully static type.

Reply:
Thanks, I see what you mean now; this is fine.